src/bosh_azure_cpi/lib/cloud/azure/restapi/azure_client.rb

```diff
@@ -108,6 +108,11 @@ class AzureClient
     # Please add the key into this list if you want to redact its value in request body.
     CREDENTIAL_KEYWORD_LIST = %w[adminPassword client_secret customData].freeze
 
+    CACHE_DIR = '/var/vcap/sys/run/azure_cpi'.freeze
```

Comment:
I was concerned about a hardcoded caching dir and `bosh create-env`, where `/var/vcap` may not exist. But it does only cache if the directory exists, so it should be fine.

Reply:
Not sure if there is a programmatic way to get this directory path within the cpi. The `create-env` case should work with this implementation (did not validate it though).